File: 2025/day01/part1.cc
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
  ifstream file("input.txt");

  string line;

  int start = 50;
  int ans = 0;
  while (getline(file, line)) {
    char dir;
    int amount;

    sscanf(line.c_str(), "%c%d", &dir, &amount);
    if (dir == 'L') {
      start -= amount;
    } else {
      start += amount;
    }
    start = ((start % 100) + 100) % 100;
    if (start == 0) {
      ans++;
    }
  }
  cout << ans;

  file.close();
  return 0;
}

File: 2025/day01/part1.py
start = 50
ans = 0

with open("input.txt") as file:
    for line in file:
        dir = line[0]
        amount = int(line[1:])

        if dir == "L":
            start -= amount
        else:
            start += amount

        start = start % 100
        if start == 0:
            ans += 1

print(ans)
